dali/operators/image/paste/multipaste.h

```diff
@@ -87,112 +115,295 @@ class MultiPasteOp : public StatelessOperator<Backend> {
+                 make_string("The `", arg_name, "` must be float in [0, 1] range, got ", factor,
+                             " for region at index ", paste_idx,
+                             " pasted into output sampe at index ", out_sample_idx, "."));
+  }
 
+  void AcquireInIdxs(const Workspace &ws) {
+    auto nsamples = ws.GetInputShape(0).num_samples();
+    if (in_idx_.HasExplicitValue()) {
+      DALI_ENFORCE(ws.NumInput() == 1,
+                   make_string("If the `in_ids` is specified, the operator accepts exactly one "
+                               "positional input batch. Got: ",
+                               ws.NumInput(), " inputs."));
+      in_idx_.Acquire(spec_, ws, nsamples);
+      for (int i = 0; i < nsamples; i++) {
+        auto paste_count = GetPasteCount(ws, i);
+        for (int j = 0; j < paste_count; j++) {
+          auto paste_idx = in_idx_[i].data[j];
+          DALI_ENFORCE(
+              0 <= paste_idx && paste_idx < nsamples,
+              make_string("The `in_idx` must be in range [0, .., batch_size - 1]. Got in_idx: ",
+                          paste_idx, ". Input batch size is: ", nsamples, "."));
+        }
+      }
+    }
+  }
+
+  // Get, for the given output sample, the number of channels. It's inferred
+  // from the pasted regions, or -1 if there are no pasted regions.
+  inline int InferChannelsNum(const Workspace &ws, int out_sample_idx) {
+    int num_channels = -1;
+    for (int paste_idx = 0; paste_idx < GetPasteCount(ws, out_sample_idx); paste_idx++) {
+      const auto &sample = GetPasteSample(ws, out_sample_idx, paste_idx);
+      int input_channels = sample.shape()[spatial_ndim_];
+      DALI_ENFORCE(paste_idx == 0 || num_channels == input_channels,
+                   make_string("All regions pasted into given output sample must have the same "
+                               "number of channels. Got different number of channels: ",
+                               num_channels, ", ", input_channels, " for output sample at index ",
+                               out_sample_idx, "."));
+      num_channels = input_channels;
+    }
+    return num_channels;
+  }
+
+  void AcquireSetOutputShape(TensorListShape<> &out_shape, const Workspace &ws, int ndim) {
+    const auto &in_shape = ws.GetInputShape(0);
+    int nsamples = in_shape.num_samples();
+    out_shape.resize(nsamples, ndim);
+
+    if (nsamples == 0) {
+      return;
+    }
+
+    TensorShape<2> ref_shape = in_shape[0].first(spatial_ndim_);
+    // validate sources of the shape
+    if (output_size_.HasExplicitValue()) {
+      output_size_.Acquire(spec_, ws, nsamples, ArgValue_EnforceUniform);
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        const auto &sample_out_shape = output_size_[sample_idx];
+        DALI_ENFORCE(sample_out_shape.num_elements() >= spatial_ndim_,
+                     make_string("The `output_size` should be a tuple (HW), got ",
```

Comment:
```suggestion
                     make_string("The `output_size` should be a tuple (H, W), got ",
```
Nitpick, I just like it more when tuple notation is used for tuple-like